src/datadog/grok/parse_grok.rs

```diff
@@ -5,37 +5,61 @@ use super::{
 use crate::path::parse_value_path;
 use crate::value::{ObjectMap, Value};
 use std::collections::BTreeMap;
-use tracing::{error, warn};
+use tracing::error;
 
+/// Errors which cause the Datadog grok algorithm to stop processing and not return a parsed result.
 #[derive(thiserror::Error, Debug, PartialEq, Eq)]
-pub enum Error {
-    #[error("failed to apply filter '{}' to '{}'", .0, .1)]
-    FailedToApplyFilter(String, String),
+pub enum FatalGrokError {
     #[error("value does not match any rule")]
     NoMatch,
-    #[error("failure occurred during match of the pattern against the value: '{}'", .0)]
-    FailedToMatch(String),
+    #[error("failure during regex engine runtime for match of the pattern against the value.")]
+    RegexEngineError,
+}
+
+/// Errors that do not prohibit the Datadog grok algorithm from continuing processing.
+#[derive(thiserror::Error, Debug, PartialEq, Eq)]
+pub enum RecoverableGrokError {
```

Comment:
> I would argue that `NoMatch` is not actually an error

I agree, but I don't want to derail this PR.

> In what way are the latter errors "recoverable"? 

True, most VRL functions are expected to be idempotent.

Reply:
I agree the naming is weird but I would argue that it is somewhat unavoidable without larger changes.

I attempted to explain the "recoverable" errors in the doc comment, but I can adjust that if needed out of this thread's discussion.

So, "recoverable" here means the grok parser itself, not for the caller. That is why the "Fatal" is the only one returned in the `Result` of `parse_grok`.

I'm open to alternative naming suggestions than recoverable. I considered using "warning" but that seemed confusing because this is an `Error` type.

Reply:
i.e , these are errors that `parse_grok` itself recovered from.

Reply:
Maybe `Error` vs `Failure` might make sense, I don't know.

Reply:
> So, "recoverable" here means the grok parser itself, not for the caller.

Sounds like `internal` is a good qualification here.

Reply:
I'm on board for renaming `RecoverableGrokError` to `InternalGrokError` 👍

Reply:
> That makes sense to me as well. Do we need to repeat Grok in these type names?

👍 

and nope that is redundant, I'll remove it.

Thanks for that suggestion @pront